hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -422,9 +422,19 @@ private void submitConnectToClusterTask() {
                             logger.warning("No connection to cluster: " + clusterId);
                         }
 
-                        submitConnectToClusterTask();
+                        submitConnectToClusterTask(initialConnection);
                     }
                 }
+                if (initialConnection && !activeConnections.isEmpty()) {
+                    // Send state to cluster for async clients in async way.
+                    executor.submit(() -> {
+                        try {
+                            client.sendStateToCluster();
+                        } catch (Throwable e) {
+                            logger.warning("Could not send state to cluster: " + e.getMessage());
```

Comment:
Is this warning enough? Should we disconnect the client from the cluster on the failure? Or let the client operate even though the cluster state is incorrect?

Without this PR, the async client also can not be used without the cluster state being sent properly but of course it blocks. Hence, allowing the operation with incorrect state would be a new behaviour if desired to do so.

Reply:
I think the client should not continue operating in this state so a warning is not enough IMO

Reply:
Ok, I will shutdown the client as well and change log message as well as level

Reply:
Added a test related to this as well.